src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+                self._z = set(groundset).difference(self.support())
+
+            # groundset should be everything
+            if not set(groundset).issubset(self.support().union(self.zeros())):
+                raise ValueError(
+                    "every element must be either positive, negative or zero")
+            self._g = groundset
+
+        self._g = list(self._g)
+
+    def __call__(self, var):
+        """
+        Return the sign of an element in the groundset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]
+            sage: E(1)
+            0
+            sage: E(0)
+            1
+            sage: E(2)
+            -1
+        """
+        if var in self.positives():
+            return 1
+        if var in self.negatives():
+            return -1
+        if var in self.zeros():
+            return 0
+        raise ValueError("not in groundset")
+
+    def __hash__(self):
+        """
+        Return hashed string of signed subset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]
+            sage: hash(E)
+            1231571227600608182
+        """
+        fsp = frozenset(self._p)
+        fsn = frozenset(self._n)
+        fsz = frozenset(self._z)
+        return hash((fsp, fsn, fsz))
+
+    def __neg__(self):
+        """
+        Return the opposite signed subset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]; E
+            +: 0
+            -: 2
+            0: 1
+            sage: -E
+            +: 2
+            -: 0
+            0: 1
+        """
+        N = copy.copy(self)
+        N._p = self._n
+        N._n = self._p
+        return N
+
+    def __eq__(self, other):
+        """
+        Return whether two elements are equal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]
+            sage: E2 = M.elements()[1]
+            sage: E1 == E2
+            False
+            sage: E1 == E1
+            True
+        """
+        if isinstance(other, SignedSubsetElement):
+            if self._p == other._p \
+                    and self._n == other._n \
+                    and self._z == other._z:
```

Comment:
```suggestion
            if (self._p == other._p
                    and self._n == other._n
                    and self._z == other._z):
```
This is considered to be better style since you won't accidentally pick up the next line if unintended.